src/tools/importPlanGoals.js

```diff
@@ -135,26 +145,53 @@ export default async function importGoals(fileKey, region) {
             const dbGrant = await Grant.findOne({ where: { ...fullGrant }, attributes: ['id', 'recipientId'] });
             if (!dbGrant) {
               // eslint-disable-next-line no-console
-              console.log(`Couldn't find grant: ${fullGrant.number}. Exiting...`);
+              logger.error(`Couldn't find grant: ${fullGrant.number}. Exiting...`);
               throw new Error('error');
             }
             const grantId = dbGrant.id;
-            await Goal.findOrCreate({
-              where: { grantId, name: goal.name, isFromSmartsheetTtaPlan: true },
-              defaults: goal,
+            const dbGoals = await Goal.findAll({
+              where: { grantId, name: goal.name },
             });
+
+            if (dbGoals.length > 1) {
+              logger.info(`Found multiples of goal id: ${dbGoals[0].id}`);
+              logger.info(`Incoming status: ${goal.status}`);
+              logger.info(`Incoming last edited date: ${currentLastEditedDate}`);
+              logger.info(`Incoming timeframe: ${goal.timeframe}`);
+              for await (const dbgoal of dbGoals) {
+                // Unable to determine a reliable update; Skipping
+                logger.info(`Skipping updates for goal: ${dbgoal.id}`);
+                logger.info(`db goal status: ${dbgoal.status}, createdAt: ${dbgoal.createdAt}`);
+              }
+            } else if (dbGoals.length === 1) {
+              const dbGoal = dbGoals[0];
+              // update timeframe
+              await dbGoal.update(
+                {
+                  timeframe: goal.timeframe,
+                  isRttapa: 'Yes',
+                },
+                {
+                  // where: { id: dbgoal.id },
+                  individualHooks: true,
+                },
+              );
+              // determine if status needs to be updated
+              await updateStatus(goal, dbGoal);
```

Comment:
Instead of running two updates, could the status be conditionally set? This way we don't unnecessarily fill the audit log and protentional make undesired updates to the timeframe and isRttapa. As if the status is ignored, should the other updates maybe be ignored too.

Reply:
I see the audit log consideration, but the status updates should not be extensive, so the audit log will only get minimal updates. We also want to bring timeframe for the data consistency. Ensuring isRttapa is set in the same update will go along with what we discussed previously and agreed that all goals from RTTAPA will be marked so in the hub.